cmd/node-cache/app/cache_app.go

```diff
@@ -271,6 +271,7 @@ func (c *CacheApp) setupNetworking() {
 func (c *CacheApp) runPeriodic() {
 	c.exitChan = make(chan struct{}, 1)
 	tick := time.NewTicker(c.params.Interval * time.Second)
+	c.setupNetworking()
```

Comment:
This is already called in cacheApp.Init(). Is that not sufficient?

Reply:
No, because Init() skips setting up iptables and only sets up interfaces.
This means the first time we setup iptables is after the first interval of `SyncInterval`, so by default we spend 60 seconds doing nothing when we start up.

Reply:
The iptables setup was skipped in Init because we need coremain.Run() to finish before installing iptables rules to redirect traffic to node-local-dns. With the change in this PR, we could still run into that race where iptables rules got added when coreDNS was still starting up, right?

Until iptables rules are added, requests will go to the clusterDNS, so there should be no DNS downtime. How does it affect node-readiness?

Reply:
I am gating my node's readiness until node-local-dns is ready to avoid workloads being scheduled during the window where the iptable rule is inserted.

If we don't want to have a race between `coremain.Run()` and setting up iptables, would it be accetable to introduce a second flag for Initial Delay which would be a more reasonable number?

My use case is that I want to keep the SyncInterval 60s but want a faster startup

Reply:
Instead of this, can we check for the pidfile of coreDNS to be present, to infer that coreDNS has started and then run the first instance of c.setupNetworking?

https://github.com/kubernetes/dns/blob/05a4cd0f5d3b0e7a2925ff10995eb821a683103d/vendor/github.com/caddyserver/caddy/caddy.go#L478 writes a pidfile if the "pidfile" argument is provided on the commandline.

Reply:
Thanks @prameshj, this will be a more deterministic solution - I will look into it

Reply:
@prameshj Done, please review.
I've done some basic functional tests and it works well, also added a unit test.
Let me know if you have any comments